src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -2001,9 +2006,12 @@ BoundNode bind(CSharpSyntaxNode root, DiagnosticBag diagnosticBag, out Binder bi
             void rewriteAndCache(DiagnosticBag diagnosticBag)
             {
 #if DEBUG
-                if (!Compilation.NullableSemanticAnalysisEnabled)
+                if (!Compilation.IsNullableAnalysisEnabled)
```

Comment:
Why do we want to consider `IsNullableAnalysisEnabled` here? All of our other `DEBUG` checks focus only on making sure it's not explicitly disabled.

Reply:
I'm not sure why we skipped `AnalyzeBoundNodeNullability()` here previously if the project has not enabled nullability. That behavior is unchanged.

---
In reply to: [541408515](https://github.com/dotnet/roslyn/pull/49876#discussion_r541408515) [](ancestors = 541408515)